libraries/net/include/graphene/net/node_configuration.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <graphene/net/config.hpp>
+
+#include <fc/crypto/elliptic.hpp>
+#include <fc/network/ip.hpp>
+
+namespace graphene { namespace net {
+
+struct node_configuration
+{
+   node_configuration() {}
+
+   fc::ip::endpoint listen_endpoint;
```

Comment:
I don't think we need to endpoint listed here. All of the behavior to determine if the node should listen or not is handled in `application.cpp`.